sklearn/decomposition/pca.py

```diff
@@ -120,25 +121,55 @@ class PCA(BaseEstimator, TransformerMixin):
 
             n_components == min(n_samples, n_features)
 
-        if n_components == 'mle', Minka\'s MLE is used to guess the dimension
-        if ``0 < n_components < 1``, select the number of components such that
-        the amount of variance that needs to be explained is greater than the
-        percentage specified by n_components
+        if n_components == 'mle' and svd_solver == 'full', Minka\'s MLE is used
+        to guess the dimension
+        if ``0 < n_components < 1`` and svd_solver == 'full', select the number
+        of components such that the amount of variance that needs to be
+        explained is greater than the percentage specified by n_components
+        n_components cannot be equal to n_features for svd_solver == 'arpack'.
 
-    copy : bool
+    copy : bool (default False)
         If False, data passed to fit are overwritten and running
         fit(X).transform(X) will not yield the expected results,
         use fit_transform(X) instead.
 
-    whiten : bool, optional
-        When True (False by default) the `components_` vectors are divided
-        by n_samples times singular values to ensure uncorrelated outputs
-        with unit component-wise variances.
+    whiten : bool, optional (default False)
+        When True the `components_` vectors are divided by n_samples times
+        singular values to ensure uncorrelated outputs with unit component-wise
+        variances.
 
         Whitening will remove some information from the transformed signal
         (the relative variance scales of the components) but can sometime
         improve the predictive accuracy of the downstream estimators by
-        making there data respect some hard-wired assumptions.
+        making their data respect some hard-wired assumptions.
+
+    svd_solver : string (default 'auto')
+        The algorithm that runs SVD
+        if svd_solver == 'full', run exact SVD calling LAPACK solver via
+        `scipy.linalg.svd` and select the components by postprocessing
+        if svd_solver == 'arpack', run SVD truncated to n_components calling
+        ARPACK solver via `scipy.sparse.linalg.svds`. It requires strictly
+        0 < n_components < X.shape[1]
+        if svd_solver == 'randomized', run randomized SVD by the method of
+        Halko et al.
+        if svd_solver == 'auto':
+            if n_components >= 1 and n_components < .05 * min(X.shape):
+                svd_solver = 'arpack'
+            elif n_components >= 1 and n_components < .8 * min(X.shape):
+                svd_solver = 'randomized'
+            else:
+                svd_solver = 'full'
+
+    tol : float >= 0, optional (default .0)
+        Tolerance for singular values computed by svd_solver == 'arpack'.
+
+    iterated_power : int >= 0, optional (default 3)
+        Number of iterations for the power method computed by
+        svd_solver == 'randomized'.
```

Comment:
Yes but we don't expose the power iteration normalizer parameter of randomized svd in the PCA class so I don't think we should mention it here. We could expose that hyperparameter but that it is likely to make the PCA class more complex for the user for little pratical benefit as the power iteration normalizer auto heuristic seems to always perform in a close to optimal way (fast and accurate for a given value of `iterated_power`).